tests/python.bats

```diff
@@ -123,11 +128,113 @@ teardown() {
+
+  export _FLOX_USE_CATALOG_MOCK="$TESTS_DIR/catalog_responses/init/python_poetry.json"
+  run "$FLOX_BIN" init --auto-setup
+  assert_success
+  assert_output --partial "'poetry' installed"
+
+  export _FLOX_USE_CATALOG_MOCK="$TESTS_DIR/catalog_responses/init/python_poetry_zlib.json"
+  "$FLOX_BIN" install zlib
+  export _FLOX_USE_CATALOG_MOCK="$TESTS_DIR/catalog_responses/empty_responses.json"
+
+  run "$FLOX_BIN" activate -- python -m project
+  assert_success
+  assert_line "<class 'numpy.ndarray'>"
+}
+
+# bats test_tags=python:activate:pyproject:pip,catalog
+@test "catalog: flox activate works with pyproject and pip" {
+  cp -r "$TESTS_DIR"/python/single-dependency/common/* "$PROJECT_DIR/"
+  cp -r "$TESTS_DIR"/python/single-dependency/pyproject-pip/* "$PROJECT_DIR/"
+
+  export _FLOX_USE_CATALOG_MOCK="$TESTS_DIR/catalog_responses/init/python_pyproject_pip.json"
+  run "$FLOX_BIN" init --auto-setup
+  assert_success
+
+  export _FLOX_USE_CATALOG_MOCK="$TESTS_DIR/catalog_responses/init/python_pyproject_pip_zlib.json"
+  "$FLOX_BIN" install zlib
+  export _FLOX_USE_CATALOG_MOCK="$TESTS_DIR/catalog_responses/empty_responses.json"
+
+  run "$FLOX_BIN" activate -- python -m project
+  assert_success
+  assert_line "<class 'numpy.ndarray'>"
+}
+
+# bats test_tags=python:activate:requirements,catalog
+@test "catalog: flox activate works with requirements.txt and pip" {
+  cp -r "$TESTS_DIR"/python/single-dependency/common/* "$PROJECT_DIR/"
+  cp -r "$TESTS_DIR"/python/single-dependency/requirements/* "$PROJECT_DIR/"
+
+  export _FLOX_USE_CATALOG_MOCK="$TESTS_DIR/catalog_responses/init/python_requirements.json"
+  run "$FLOX_BIN" init --auto-setup
+  assert_success
+
+  export _FLOX_USE_CATALOG_MOCK="$TESTS_DIR/catalog_responses/init/python_requirements_zlib.json"
+  "$FLOX_BIN" install zlib
+  export _FLOX_USE_CATALOG_MOCK="$TESTS_DIR/catalog_responses/empty_responses.json"
+
+  run "$FLOX_BIN" activate -- python -m project
+  assert_success
+  assert_line "<class 'numpy.ndarray'>"
+}
+
+# bats test_tags=init:python:auto-setup,init:python:auto-setup:bash,catalog
+@test "catalog: verify auto-setup Python venv activation: bash" {
+  OWNER="owner"
+  NAME="name"
+  echo "requests" > requirements.txt
+  [ ! -e .flox ] || "$FLOX_BIN" delete -f
+  export _FLOX_USE_CATALOG_MOCK="$TESTS_DIR/catalog_responses/init/python_requests.json"
+  "$FLOX_BIN" init --auto-setup --name "$NAME"
+  export _FLOX_USE_CATALOG_MOCK="$TESTS_DIR/catalog_responses/empty_responses.json"
+  FLOX_SHELL="bash" run "$FLOX_BIN" activate -- type deactivate
+  assert_success
+  assert_line --partial "deactivate is a function"
+}
+
+# bats test_tags=init:python:auto-setup,init:python:auto-setup:zsh,catalog
+@test "catalog: verify auto-setup Python venv activation: zsh" {
   OWNER="owner"
   NAME="name"
   echo "requests" > requirements.txt
   [ ! -e .flox ] || "$FLOX_BIN" delete -f
+  export _FLOX_USE_CATALOG_MOCK="$TESTS_DIR/catalog_responses/init/python_requests.json"
```

Comment:
nit: it would probably be a better habit to just set vars for single commands rather than exporting and having to re-export `empty_responses.json`

Reply:
Fixed